Update ci.yml

PieterTack opened the pull request.

PieterTack committed:
Update ci.yml

PieterTack commented on .github/workflows/ci.yml:
```diff
@@ -199,29 +180,12 @@ jobs:
         brew update
         brew upgrade || true
-        brew install hdf5 git autoconf automake libtool curl wget pkg-config gsl numpy meson ninja
+        brew install hdf5 git libtool curl wget pkg-config gsl numpy meson ninja
```
I suppose this is ok? Autoconf and automake not needed anymore? What about libtool, what is that used for?

tschoonj replied:
You can also safely get rid of `libtool`, which is also part of the GNU autotools

PieterTack committed:
Delete configure.ac

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete m4 directory

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack committed:
Delete Makefile.am

PieterTack commented:
So as far as I understood, meson never uses Automake.am files, so I went through all directories and removed them. Also configure.ac shouldn't be used by meson, so got rid of that.
Then I made some changes to the ci.yml... There are a few lines of code that I'm not sure about.... I added some comments there.

PieterTack committed:
Update ci.yml

PieterTack commented:
In 'ci.yml', in build msys2/setup-msys2@v2; is it safe to remove make etc here as well? I guess not, as it is then used to make xraylib etc? Or can we do this differently on msys2 as well?

PieterTack committed:
Google Analytics removal

PieterTack committed:
Last google analytics line removal

PieterTack committed:
Installation dox adjusted for autotools removal

PieterTack committed:
ci optimise: attempt1 - WIP

PieterTack committed:
ci optimise: attempt2

PieterTack committed:
ci optimise Attempt3 - WIP

PieterTack committed:
ci optimise - Attempt 4 - WIP

PieterTack committed:
ci optimise - Attempt5 - WIP

PieterTack committed:
ci optimise - Attempt6 - Should be getting somewhere?

PieterTack committed:
ci optimise - Attempt7 - pro/regress

PieterTack committed:
ci optimise - Attempt8 - revert back to ninja
openmp appears to give issues using meson compile etc

PieterTack committed:
ci trial-and-error - attempt 9
the openmp issue in ubuntu:latest stumps me. Wasn't an issue before?

PieterTack commented:
Status note:
- visual studio: "found pkg-config 'C:\\Strawberry\\perl\\bin\\pkg-config.BAT' but it is Strawberry Perl and thus broken. Ignoring..."
--> how does one properly install pkg-config on windows system/visualstudio?

- ubuntu:latest: " OpenMP found but omp.h missing." --> somehow meson setup does not find omp.h... From what I find I should add the -fopenmp flag during compilation, but as this isn't the compilation step yet I'm not understanding the problem at hand. Probably a path is missing somewhere? But why, as I think before this all this wasn't an issue for ubuntu systems?

PieterTack committed:
ci visualstudio guesses - Attempt 10

PieterTack committed:
nuget install? - Attempt 11

PieterTack committed:
Brew in visualstudiO?

tschoonj committed:
Try fixing ubuntu clang build

tschoonj committed:
visual studio: install xraylib

tschoonj committed:
Use meson's builtin cython support

tschoonj committed:
Try fixing powershell env variable expansion

tschoonj committed:
Fixes

tschoonj committed:
More Fixes

tschoonj committed:
More Fixes

tschoonj committed:
More Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Add support for HDF5 1.14

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Dump deps

tschoonj committed:
Dump deps

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Use miniconda for visual-studio ci

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Run dumpbin

tschoonj committed:
Install vcredist-all

tschoonj committed:
Set CONDA_DLL_SEARCH_MODIFICATION_ENABLE

tschoonj committed:
Try using os.add_dll_directory

tschoonj committed:
Switch back to using vcpkg based approach

tschoonj committed:
Fix

tschoonj committed:
Enable more visual studio builds

tschoonj committed:
Fixes

tschoonj committed:
Fixes

tschoonj committed:
Cleanup

tschoonj commented:
@PieterTack I got the visual studio build working after a lot of trial and error. Would be great if you could review if you have a moment.

tschoonj commented:
Note that this PR already includes the changes necessary to support HDF5 1.14, as this version is in use by both msys2 and vcpkg.
